structuralcodes/codes/mc2010/_concrete_torsion.py

```diff
@@ -0,0 +1,172 @@
+        a_k: Can be found in figure 7.3-18
+
+    Returns:
+        The shear force that will ocurre due to torsion force."""
+    return t_ed * z_i / (2 * a_k)
+
+
+def t_rd_max(
+    f_ck: float,
+    d_k: float,
+    a_k: float,
+    theta: float,
+    approx_lvl: int,
+    z: float,
+    E_s: float,
+    As: float,
+    loads: dict,
+    gamma_c: float = 1.5,
+) -> float:
+    """The maximum allowed torsion allowed
+
+    fib Model Code 2010, eq. (7.3-56)
+
+    Args:
+        f_ck (float): Characteristic strength in MPa
+        gamma_c (float): Concrete safety factor
+        d_k (float): Is the diameter in the smalest circel in the cross section
+        a_k: Can be found in figure 7.3-18
+        theta (float): Inclitaniton of the compression stressfield in degrees
+        approx_lvl (int): Approximation method for cocrete with reinforcement
+        z (float): distances between the centerline of the
+        compressive chord and the reinforcement in mm
+        E_s (float): The E_s-modulus to the materialb in MPa
+        As (float): The cross-section reinforcement in mm^2
+        loads (dictionary) The given loads in a dictionary:
+            Med (float): The positive moment working on the material in Nmm
+            Ved (float): The positive shear force working on the material in N
+            Ned (float): The normal force working on the material in N with
+            positive sign for tension and negative sign for compression
+            delta_E (float): The eccentricity of the axial load due to
+            imperfection in the construction with distance in mm as a positive
+            value
+        gamma_c (float): Concrete safety factor
+
+
+    Return:
+        The maximum allowed torsion allowed
+    """
+    t_ef = d_k / 8
+
+    if approx_lvl == 3:
+        k_epsilon = 0.55
+    elif approx_lvl == 4:
+        epsilon_1 = epsilon_x(E_s, As, z, loads) + (
+            epsilon_x(E_s, As, z, loads) + 0.002
+        ) * ((1 / tan(theta * pi / 180)) ** 2)
+        k_epsilon = min(1 / (1.2 + 55 * epsilon_1), 0.65)
+    elif approx_lvl == 5:
+        theta_min = 20 + 10000 * epsilon_x(E_s, As, z, loads)
+        epsilon_1 = epsilon_x(E_s, As, z, loads) + (
+            epsilon_x(E_s, As, z, loads) + 0.002
+        ) * ((1 / tan(theta_min * pi / 180)) ** 2)
+        k_epsilon = min(1 / (1.2 + 55 * epsilon_1), 0.65)
+    k_c = eta_fc(f_ck) * k_epsilon
+    result = (
+        k_c
+        * f_ck
+        * t_ef
+        * 2
+        * a_k
+        * sin(theta * pi / 180)
+        * cos(theta * pi / 180)
+        / gamma_c
+    )
+
+    return result
+
+
+def t_rd(  # pylint: disable=r0801
+    t_ed: float,
+    approx_lvl: int,
+    fck: float,
+    bw: float,
+    theta: float,
+    z: float,
+    E_s: float,
+    As: float,
+    loads: dict,
+    d_k: float,
+    a_k: float,
+    alpha: float = 90.0,
+    gamma_c: float = 1.5,
+) -> bool:
+    """Checks if the combination of torstion ans shear is ok
```

Comment:
Typos:
torstion -> torsion
ans -> and
choosen -> chosen
inclitaniton -> inclination
E_s-modulus -> E-modulus
materialb -> material
smalest -> smallest
criteria -> criterion (singular)

Reply:
Done